torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -936,57 +936,38 @@ at::Tensor XLANativeFunctions::bitwise_and(const at::Tensor& self,
       });
 }
 
-at::Tensor& XLANativeFunctions::bitwise_not_out(const at::Tensor& self,
-                                                at::Tensor& out) {
+at::Tensor XLANativeFunctions::bitwise_not(const at::Tensor& self) {
   XLA_FN_COUNTER("xla::");
-  XLATensor out_tensor = bridge::GetXlaTensor(out);
   XLATensor self_tensor = bridge::GetXlaTensor(self);
-  XLATensor::bitwise_not_out(out_tensor, self_tensor);
-  return out;
+  return bridge::AtenFromXlaTensor(XLATensor::bitwise_not(self_tensor));
 }
 
-at::Tensor& XLANativeFunctions::bitwise_or_out(const at::Tensor& self,
-                                               const at::Scalar& other,
-                                               at::Tensor& out) {
+at::Tensor XLANativeFunctions::bitwise_or(const at::Tensor& self,
+                                          const at::Scalar& other) {
   XLA_FN_COUNTER("xla::");
-  CheckBinaryOpTypePromotion(out, self, other);
-  XLATensor out_tensor = bridge::GetXlaTensor(out);
-  XLATensor::bitwise_or_out(out_tensor, bridge::GetXlaTensor(self), other);
-  return out;
+  return bridge::AtenFromXlaTensor(
+      XLATensor::bitwise_or(bridge::GetXlaTensor(self), other));
 }
 
-at::Tensor& XLANativeFunctions::bitwise_or_out(const at::Tensor& self,
-                                               const at::Tensor& other,
-                                               at::Tensor& out) {
+at::Tensor XLANativeFunctions::bitwise_or(const at::Tensor& self,
+                                          const at::Tensor& other) {
   XLA_FN_COUNTER("xla::");
-  DoBinaryOpOut(
-      self, other, out,
-      [&](const XLATensor& xself, const XLATensor& xother, XLATensor& xout) {
-        XLATensor::bitwise_or_out(xout, xself, xother);
-      });
-  return out;
+  return bridge::AtenFromXlaTensor(XLATensor::bitwise_or(
```

Comment:
We used to need `DoBinaryOpOut` or `DoBinaryOp` to guarantee the output `logical_element_type` being set correctly. I guess when we moved to codegen those ops it won;t be necessary since we have the meta kernel to compute the output shape. However before we actually codegen these ops, I don't know if getting rid fo `DoBinaryOp` will result in any kind of result type mismatch issue.

Reply:
Oh right, thanks!

```
at::Tensor DoBinaryOp(const at::Tensor& self, const at::Tensor& other,
                      const B& bin_op) {
  at::ScalarType dtype = at::result_type(self, other);
  std::pair<XLATensor, XLATensor> operands =
      GetBinaryOperands(self, UnwrapNumber(other, dtype));
  XLATensor result = bin_op(operands.first, operands.second, dtype);
  return bridge::AtenFromXlaTensor(result);
}
```
In particular, looks like `DoBinaryOp` is needed to properly handle type promotion when dealing with wrapped scalar tensor inputs? Originally I just saw that `DoBinaryOpOut` has an extra assert for the output dtype, that `DoBinaryOp` doesn't need. I'll add it back.